contracts/HolographOperator.sol

```diff
@@ -431,65 +436,99 @@ contract HolographOperator is Admin, Initializable, IHolographOperator {
+        uint64(packed >> 16),
+        [
+          uint16(packed >> 160),
+          uint16(packed >> 144),
+          uint16(packed >> 128),
+          uint16(packed >> 112),
+          uint16(packed >> 96)
+        ]
+      );
   }
 
-  function _popOperator(uint256 pod, uint256 operatorIndex) private {
-    if (operatorIndex > 0) {
-      unchecked {
-        address operator = _operatorPods[pod][operatorIndex];
-        // remove operator pod reference
-        _bondedOperators[operator] = 0;
-        uint256 lastIndex = _operatorPods[pod].length - 1;
-        if (lastIndex != operatorIndex) {
-          _operatorPods[pod][operatorIndex] = _operatorPods[pod][lastIndex];
-        }
-        delete _operatorPods[pod][lastIndex];
-        _operatorPods[pod].pop();
-      }
-    }
+  function getPodOperators(uint256 pod) external view returns (address[] memory operators) {
+    require(_operatorPods.length >= pod, "HOLOGRAPH: pod does not exist");
+    operators = _operatorPods[pod - 1];
   }
 
-  function _bridge() private view returns (address bridge) {
-    assembly {
-      bridge := sload(_bridgeSlot)
+  function getPodOperators(
+    uint256 pod,
+    uint256 index,
+    uint256 length
+  ) external view returns (address[] memory operators) {
+    require(_operatorPods.length >= pod, "HOLOGRAPH: pod does not exist");
+    // decrease by one for easy code usage
+    pod--;
+    uint256 supply = _operatorPods[pod].length;
+    if (index + length > supply) {
+      length = supply - index;
+    }
+    operators = new address[](length);
+    for (uint256 i = 0; i < length; i++) {
+      operators[i] = _operatorPods[pod][index + i];
     }
   }
 
-  function _holograph() private view returns (address holograph) {
-    assembly {
-      holograph := sload(_holographSlot)
-    }
+  function getPodBondAmount(uint256 pod) external view returns (uint256 base, uint256 current) {
+    base = _getBaseBondAmount(pod - 1);
+    current = _getCurrentBondAmount(pod - 1);
   }
 
-  function _interfaces() private view returns (IInterfaces interfaces) {
-    assembly {
-      interfaces := sload(_interfacesSlot)
-    }
+  function getBondedPod(address operator) external view returns (uint256 pod) {
+    return _bondedOperators[operator];
   }
 
-  function _registry() private view returns (address registry) {
-    assembly {
-      registry := sload(_registrySlot)
+  // add top-up option
+
+  function bondUtilityToken(
+    address operator,
+    uint256 amount,
+    uint256 pod
+  ) external {
+    require(_bondedOperators[operator] == 0, "HOLOGRAPH: operator is bonded");
+    unchecked {
+      uint256 current = _getCurrentBondAmount(pod - 1);
+      require(current <= amount, "HOLOGRAPH: bond amount too small");
+      // subtract difference and only keep bond amount
+      if (_operatorPods.length < pod) {
+        for (uint256 i = _operatorPods.length; i <= pod; i++) {
+          _operatorPods.push([address(0)]);
```

Comment:
Why pushing zero address here?

Reply:
A pod cannot be empty, this way it allows for a zero address to be selected if the pod is selected. Zero address allows anyone to operate and is kind of like a winning the lotto scenario for operators.

Reply:
ahh okay that's cool